src/safe_allocator.h

```diff
@@ -46,28 +43,37 @@ class SafeAllocator : public Allocator {
 
     void
     Deallocate(void* p) override {
-        raw_allocator_->Deallocate(p);
+        allocator_->Deallocate(p);
     }
 
     void*
     Reallocate(void* p, size_t size) override {
-        auto ret = raw_allocator_->Reallocate(p, size);
+        auto ret = allocator_->Reallocate(p, size);
         if (not ret) {
             throw std::bad_alloc();
         }
         return ret;
     }
     Allocator*
     GetRawAllocator() {
-        return raw_allocator_;
+        return allocator_;
     }
 
 public:
-    ~SafeAllocator() override = default;
+    ~SafeAllocator() override {
+        if (owner_) {
+            delete allocator_;
+        }
+    };
```

Comment:
when should SafeAllocator release the allocator_?

Reply:
set owner flag for default allocator